src/main/java/org/jboss/logmanager/BootstrapConfiguration.java

```diff
@@ -0,0 +1,444 @@
+     */
+    public BootstrapConfiguration setHandler(final Handler handler) {
+        Assert.checkNotNullParam("handler", handler);
+        return setHandler(new Supplier<Handler>() {
+            @Override
+            public Handler get() {
+                return handler;
+            }
+        });
+    }
+
+    /**
+     * Sets the handler to use for logging messages to if the JVM shuts down before the
+     * {@link LogContext#configurationComplete()} is invoked. Note the handler will lazily be initialized only if
+     * the shutdown hook is invoked.
+     *
+     * @param handler the handler to use
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration setHandler(final Supplier<? extends Handler> handler) {
+        this.handler = Assert.checkNotNullParam("handler", handler);
+        return this;
+    }
+
+    /**
+     * A shortcut to supplying a {@link ConsoleHandler} which will be lazily loaded if the JVM shuts down before the
+     * {@link LogContext#configurationComplete()} is invoked.
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration useConsoleHandler() {
+        return setHandler(new Supplier<Handler>() {
+            @Override
+            public Handler get() {
+                return new ConsoleHandler(new PatternFormatter(PATTERN));
+            }
+        });
+    }
+
+    /**
+     * A shortcut to supplying a {@link FileHandler} which will be lazily loaded if the JVM shuts down before the
+     * {@link LogContext#configurationComplete()} is invoked.
+     *
+     * @param path the path to the file to write messages to
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration useFileHandler(final String path) {
+        return useFileHandler(new File(Assert.checkNotNullParam("path", path)));
+    }
+
+    /**
+     * A shortcut to supplying a {@link FileHandler} which will be lazily loaded if the JVM shuts down before the
+     * {@link LogContext#configurationComplete()} is invoked.
+     *
+     * @param path the path to the file to write messages to
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration useFileHandler(final Path path) {
+        return useFileHandler(Assert.checkNotNullParam("path", path).toFile());
+    }
+
+    /**
+     * A shortcut to supplying a {@link FileHandler} which will be lazily loaded if the JVM shuts down before the
+     * {@link LogContext#configurationComplete()} is invoked.
+     *
+     * @param path the path to the file to write messages to
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration useFileHandler(final File path) {
+        Assert.checkNotNullParam("path", path);
+        return setHandler(new Supplier<Handler>() {
+            @Override
+            public Handler get() {
+                try {
+                    return new FileHandler(new PatternFormatter(PATTERN), path);
```

Comment:
We probably should create this with `append` being `true`. Different log contexts may overwrite each other.